client/state/sites/reducer.js

```diff
@@ -108,12 +112,13 @@ export function items( state = {}, action ) {
 			);
 
 		case SITE_DELETE_RECEIVE:
-			return omit( state, action.siteId );
-
 		case JETPACK_DISCONNECT_RECEIVE:
-			return omit( state, action.siteId );
+			return state !== null ? omit( state, action.siteId ) : state;
```

Comment:
(non-blocking, optional) Just for consistency I'd find this easier to scan if we just repeated the usual bail condition:
```javascript
if ( state === null ) {
 return state;
}
```